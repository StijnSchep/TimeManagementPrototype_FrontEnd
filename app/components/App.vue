<template>
  <Page class="body" actionBarHidden="true">
    <Image src="~/assets/images/nos-managementapp-logo.png" class="logo" />
  </Page>
</template>

<script>
import Dashboard from "./dashboard/Dashboard";
import LoginPage from "./login/LoginPage";
import axios from "axios";
import { getString } from "tns-core-modules/application-settings";
import * as dialogs from "tns-core-modules/ui/dialogs";
import {
  FingerprintAuth
} from "nativescript-fingerprint-auth";

var fingerprintAuth = new FingerprintAuth();

export default {
  components: {
    Dashboard
  },
  data() {
    return {
      domains: []
    };
  },
  methods: {
    goToLogin() {
      this.$navigateTo(LoginPage, {
        clearHistory: true,
        transition: {
          name: "fade",
          duration: 400
        },
        props: {
          domains: this.domains
        }
      });
    },
    goToDashboard() {
      this.$navigateTo(Dashboard, {
        clearHistory: true,
        transition: {
          name: "fade",
          duration: 400
        }
      });
    }
  },
  mounted() {
    var usefingerprint = getString("usetouch");

    if (getString("store") && getString("store") != "") {
      fingerprintAuth.available().then(function(avail) {
        if (avail.any && avail.touch && usefingerprint === "true") {
          fingerprintAuth
            .verifyFingerprint({
              message: "Scan vingerafdruk",
              useCustomAndroidUI: false
            })
            .then(enteredPassword => {
              if (enteredPassword === undefined) {
                this.$store.commit("auth/load");
                setTimeout(() => {
                  this.goToDashboard();
                }, 2000);
              } else {
                this.$store.commit("auth/load");
                setTimeout(() => {
                  this.goToDashboard();
                }, 2000);
              }
            })
            .catch(err => {
              this.$store.commit("auth/load");
              setTimeout(() => {
                this.goToDashboard();
              }, 2000);
            });
        } else {
          this.$store.commit("auth/load");
          setTimeout(() => {
            this.goToDashboard();
          }, 2000);
        }
      });
      this.$store.commit("auth/load");
      setTimeout(() => {
        this.goToDashboard();
      }, 2000);
    } else {
      axios
        .get("https://nostrapersoneelsapi.herokuapp.com/api/company/domains")
        .then(
          res => {
            res.data.result.forEach(domain => {
              this.domains.push(domain.Domain);
            });

            setTimeout(() => {
              this.goToLogin();
            }, 1500);
          },
          error => {
            dialogs.alert({
              title: "Error",
              message: "Kon geen verbinding maken met API",
              okButtonText: "Ok!"
            });
          }
        );
    }
  }
};
</script>

<style scoped>
.body {
  background-image: url("~/assets/images/SplashScreen.jpg");
  background-repeat: no-repeat;
  background-position: center top;
  background-size: cover;
}

.logo {
  margin: 50px;
}
</style>
