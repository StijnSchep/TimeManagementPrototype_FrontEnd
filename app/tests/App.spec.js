import { shallowMount } from "@vue/test-utils";
import App from "../components/App";


jest.mock("tns-core-modules/application-settings", () => ({
  getString: jest.fn()
}));

jest.mock('nativescript-fingerprint-auth', () => ({
  FingerprintAuth: jest.fn(),
  getString: jest.fn(() => "")
}));

jest.mock('tns-core-modules/application', () => ({
  android: {
    on: jest.fn()
  },
  AndroidApplication: {
    activityBackPressedEvent: {}
  }
}));

jest.mock("tns-core-modules/ui/dialogs", () => ({
  alert: jest.fn(() => Promise.resolve())
}));

describe("App.vue", () => {
  it("mounts and renders", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.html()).toBeTruthy();
  });

  test("renders correctly", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.element).toMatchSnapshot();
  });

  it("is a Vue instance", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("Expect component to exist", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper).toBeTruthy();
  });

  it("Expect component to have Page element", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.find("Page")).toBeTruthy();
  });

  it("Expect component to have FlexboxLayout element", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.find("FlexboxLayout")).toBeTruthy();
  });

  it("Expect component to have template element", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.find("template")).toBeTruthy();
  });

  it("Expect component to have script element", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.find("script")).toBeTruthy();
  });

  it("Expect component to have style element", async () => {
    const wrapper = shallowMount(App);
    await expect(wrapper.find("style")).toBeTruthy();
  });
});
