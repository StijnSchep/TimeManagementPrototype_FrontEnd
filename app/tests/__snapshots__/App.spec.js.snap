// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders correctly 1`] = `
<page-stub
  actionbarhidden="true"
  class="body"
>
  <image
    class="logo"
    src="~/assets/images/nos-managementapp-logo.png"
  />
</page-stub>
`;
